crates/types/src/joints.rs

```diff
@@ -73,57 +64,69 @@ impl Sub for HeadJoints {
     }
 }
 
-#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize, SerializeHierarchy)]
-pub struct ArmJoints {
-    pub shoulder_pitch: f32,
-    pub shoulder_roll: f32,
-    pub elbow_yaw: f32,
-    pub elbow_roll: f32,
-    pub wrist_yaw: f32,
-    pub hand: f32,
+impl Mul<f32> for HeadJoints<f32> {
+    type Output = HeadJoints<f32>;
+
+    fn mul(self, right: f32) -> Self::Output {
+        Self::Output {
+            yaw: self.yaw * right,
+            pitch: self.pitch * right,
+        }
+    }
 }
 
-impl ArmJoints {
-    pub fn mirrored(self) -> Self {
-        Self {
-            shoulder_pitch: self.shoulder_pitch,
-            shoulder_roll: -self.shoulder_roll,
-            elbow_yaw: -self.elbow_yaw,
-            elbow_roll: -self.elbow_roll,
-            wrist_yaw: -self.wrist_yaw,
-            hand: self.hand,
+impl Div<f32> for HeadJoints<f32> {
+    type Output = HeadJoints<f32>;
+
+    fn div(self, right: f32) -> Self::Output {
+        Self::Output {
+            yaw: self.yaw / right,
+            pitch: self.pitch / right,
         }
     }
+}
 
-    pub fn fill(value: f32) -> Self {
+impl HeadJoints<f32> {
+    pub fn mirrored(self) -> Self {
         Self {
-            shoulder_pitch: value,
-            shoulder_roll: value,
-            elbow_yaw: value,
-            elbow_roll: value,
-            wrist_yaw: value,
-            hand: value,
+            yaw: -self.yaw,
+            pitch: self.pitch,
         }
     }
 }
 
-impl Mul<f32> for ArmJoints {
-    type Output = ArmJoints;
-
-    fn mul(self, scale_factor: f32) -> Self::Output {
-        Self::Output {
-            shoulder_pitch: self.shoulder_pitch * scale_factor,
-            shoulder_roll: self.shoulder_roll * scale_factor,
-            elbow_yaw: self.elbow_yaw * scale_factor,
-            elbow_roll: self.elbow_roll * scale_factor,
-            wrist_yaw: self.wrist_yaw * scale_factor,
-            hand: self.hand * scale_factor,
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize, SerializeHierarchy)]
+#[serialize_hierarchy(bound = "T: SerializeHierarchy + Serialize, for<'de> T: Deserialize<'de>")]
+pub struct ArmJoints<T = f32> {
+    pub shoulder_pitch: T,
+    pub shoulder_roll: T,
+    pub elbow_yaw: T,
+    pub elbow_roll: T,
+    pub wrist_yaw: T,
+    pub hand: T,
+}
+
+impl<T> ArmJoints<T>
+where
+    T: Clone,
```

Comment:
Is `Clone` a `Copy` for `f32`? Or do we have a performance issue here?

Reply:
?!?! what should it do differently? There should never be a difference of clone and copy. `Copy` is just a marker trait for the compiler and requires `Clone` to be more strict.

Reply:
Ok